Add S2SA nems.configure template

WalterKolczynski-NOAA opened the pull request:
**Description**
Most of the configuration was setup to allow S2SA, but there was no appropriate nems.configure template. A new one was created by starting with the S2SWA template and removing the wave portions. This functionality will support coupled DA development.

The experiment setup documentation for cycled mode had not been updated as coupled capability was added, so that has now been updated. Also added notes to both the cycled and forecast-only section that ATMW is currently not working.

Also added two UFSDA scripts that were missing from the git ignore list.

Fixes #1592

**Type of change**

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

**How Has This Been Tested?**

~~- [ ] S2SA cycled test on Orion~~
- [x] S2SA forecast-only
  
**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] Any dependent changes have been merged and published

WalterKolczynski-NOAA committed:
Add S2SA nems.configure template

Most of the configuration was setup to allow S2SA, but there was no
appropriate nems.configure template. A new one was created by
starting with the S2SWA template and removing the wave portions.

Also added two UFSDA scripts that were missing from the git ignore
list.

Fixes #1592

WalterKolczynski-NOAA committed:
Update documentation for cycled mode setup

The experiment setup documentation for cycled mode had not been
updated as coupled capability was added. Also added notes to both
the cycled and forecast-only section that ATMW is currently not
working.

WalterKolczynski-NOAA commented:
Testing is waiting on an unrelated IC issue to be resolved.

aerorahul reviewed (approved):
makes sense to me.

aerorahul commented:
@junwang-noaa would you please comment on the `nems.configure` template for the S2SA use case?

WalterKolczynski-NOAA commented:
With the gocart emissions updated, having an issue with the forecast during (seemingly) the first GOCART loop:

```
0: PASS: fcstRUN phase 1, n_atmsteps =                0 time is        17.056236
 0: UFS Aerosols: Advancing from 2021-03-23T12:00:00 to 2021-03-23T12:20:00
 4: pe=00004 FAIL at line=00827    SU2G_GridCompMod.F90                     <status=29>
 4: pe=00004 FAIL at line=00685    SU2G_GridCompMod.F90                     <status=29>
 4: pe=00004 FAIL at line=01806    MAPL_Generic.F90                         <status=29>
 4: pe=00004 FAIL at line=00167    BaseProfiler.F90                         <Timer <GOCART2G> does not match start timer <SU>>
 4: pe=00004 FAIL at line=01824    MAPL_Generic.F90                         <status=1>
 4: pe=00004 FAIL at line=00161    Aerosol_GridComp.F90                     <Failed to run child component>
 4: pe=00004 FAIL at line=01806    MAPL_Generic.F90                         <status=1>
 4: pe=00004 FAIL at line=01192    MAPL_CapGridComp.F90                     <status=1>
 4: pe=00004 FAIL at line=01144    MAPL_CapGridComp.F90                     <status=1>
 4: pe=00004 FAIL at line=00812    MAPL_CapGridComp.F90                     <status=1>
 4: pe=00004 FAIL at line=00952    MAPL_CapGridComp.F90                     <status=1>
```

The PET logs also have this error message:
```
20230601 162253.786 ERROR            PET00 Aerosol_Cap.F90:464 Invalid object  - Passing error in return code
20230601 162253.787 ERROR            PET00 CHM:src/addon/NUOPC/src/NUOPC_ModelBase.F90:2226 Invalid object  - Passing error in return code
20230601 162253.787 ERROR            PET00 UFS Driver Grid Comp:src/addon/NUOPC/src/NUOPC_Driver.F90:3340 Invalid object  - Phase 'RunPhase1' Run for modelComp 3 did not return ESMF_SUCCESS
20230601 162253.787 ERROR            PET00 UFS Driver Grid Comp:src/addon/NUOPC/src/NUOPC_Driver.F90:3625 Invalid object  - Passing error in return code
20230601 162253.787 ERROR            PET00 UFS Driver Grid Comp:src/addon/NUOPC/src/NUOPC_Driver.F90:3255 Invalid object  - Passing error in return code
20230601 162253.787 ERROR            PET00 UFS.F90:403 Invalid object  - Aborting UFS
```

Seems like something that should be a straightforward error, but I'm not sure what. Error in my new `nems.configure`? Issue with the ICs? This is a warm-start from ICs. @bbakernoaa @lipan-NOAA @junwang-noaa 

Working directory: `/work2/noaa/stmp/wkolczyn/RUNDIRS/S2SA_cyc/fcst.376447`
Log: `/work2/noaa/global/wkolczyn/noscrub/global-workflow/S2SA_cyc/logs/2021032312/gdasfcst.log`

bbakernoaa commented:
@WalterKolczynski-NOAA This is due to the volcanic emission dataset.  Sorry I missed this.  They are there for future years but in a different directory.  

If you modify the `SU2G_instance_SU.rc` file: Line 11

before

```bash
volcano_srcfilen: ExtData/volcanic/so2_volcanic_emissions_Carns.%y4%m2%d2.rc
```

after
```bash
volcano_srcfilen: ExtData/nexus/VOLCANO/v2021-09/%y4/so2_volcanic_emissions_Carns.%y4%m2%d2.rc 
```

I'll take blame for this one

WalterKolczynski-NOAA commented:
@bbakernoaa Even after making that change, I get the same error

Working directory: `/work2/noaa/stmp/wkolczyn/RUNDIRS/S2SA_cyc/fcst.430195`

bbakernoaa commented:
@WalterKolczynski-NOAA I can't open up the latest log file 

```/work2/noaa/global/wkolczyn/noscrub/global-workflow/S2SA_cyc/logs/2021032312/gdasfcst.log: Permission denied```

WalterKolczynski-NOAA commented:
> @WalterKolczynski-NOAA I can't open up the latest log file
> 
> `/work2/noaa/global/wkolczyn/noscrub/global-workflow/S2SA_cyc/logs/2021032312/gdasfcst.log: Permission denied`

Not sure why that one was any different, but should be fixed now.

bbakernoaa commented:
@WalterKolczynski-NOAA found the error in my suggestion 

it should include the month in the path... my fault.... sorry 

`volcano_srcfilen: ExtData/nexus/VOLCANO/v2021-09/%y4/%m2/so2_volcanic_emissions_Carns.%y4%m2%d2.rc`

WalterKolczynski-NOAA commented:
@bbakernoaa That worked, thanks.

WalterKolczynski-NOAA committed:
Update setup_expt to recognize more S2S cycled exps

setup_expt would only detect two of the possible S2S options when
staging cycled ICs.

Refs: #1592

WalterKolczynski-NOAA committed:
Change path to volcanic aerosol data

The volcanic aerosol data was moved to a new location within the
GOCART emissions directory.

Refs: #1592

WalterKolczynski-NOAA committed:
Disable enkf versions of coupled settings

The enkf versions of coupled settings (DO_OCN, etc.) do not work
properly because some settings have already been determined by that
point based on the non-enkf versions of the settings. The enkf
versions are now disabled until there is a solution (see #1692).

Refs: #1593
Refs: #1692

WalterKolczynski-NOAA committed:
Fix ocnpost detection for gfs free-forecast

The task list for the free-forecast was checking against a list of
apps that did not include S2SA to determine whether ocnpost is
included. That check is now changed to check on do_ocean rather than
looking at the app directly.

Refs: #1593

WalterKolczynski-NOAA commented:
While waiting for IC staging to be updated for cycled, tested using forecast-only mode instead.

aerorahul commented on .gitignore:
```diff
@@ -130,7 +130,9 @@ ush/global_chgres.sh
 ush/global_chgres_driver.sh
 ush/global_cycle.sh
 ush/global_cycle_driver.sh
+ush/imsfv3_scf2ioda.py
```
```suggestion
```
This file does not exist anymore.

WalterKolczynski-NOAA replied:
Removed

aerorahul commented on .gitignore:
```diff
@@ -130,7 +130,9 @@ ush/global_chgres.sh
 ush/global_cycle_driver.sh
+ush/imsfv3_scf2ioda.py
 ush/jediinc2fv3.py
+ush/letkf_create_ens.py
```
```suggestion
```
This file does not exist anymore.

WalterKolczynski-NOAA replied:
Removed.

aerorahul commented on workflow/setup_expt.py:
```diff
@@ -61,7 +61,7 @@ def fill_COMROT_cycled(host, inputs):
     do_ocean = do_ice = do_med = False
 
-    if inputs.app in ['S2S', 'S2SW']:
+    if 'S2S' in inputs.app:
```
Not sure why one would rely on an implicit wildcard when the explicit list is clearer.
But, `in` allows it, so I can't argue.

WalterKolczynski-NOAA replied:
This (mostly) mirrors how `config.base` does it (config.base technically only matches `S2S` at the beginning of the string, but that is the only place we put it).

aerorahul replied:
Works for me.

aerorahul reviewed (approved):
Looks good.  Assume the `nems.configure.cpld_aero.IN` gets tested in an `S2SA` app run.
Just 2 comments on non-existent files that are being ignored.

WalterKolczynski-NOAA committed:
Remove from ignore files that no longer exist

A couple of scripts that had been added to the git ignore list during
the course of this update have since been removed and no longer need
to be ignored.

junwang-noaa commented on ush/nems.configure.cpld_aero.IN:
```diff
@@ -0,0 +1,128 @@
+ATM_omp_num_threads:            @[atm_omp_num_threads]
+ATM_attributes::
+  Verbosity = 0
+  DumpFields = @[DumpFields]
```
Do we need this variable in GW? This might only be used for debug purpose. I think it can be set to false.

aerorahul replied:
@junwang-noaa I think we are just being consistent w/ the other `nems.configure` templates.  It is default set to `.false.` I think.  If not, we can address in a later PR.

junwang-noaa commented on ush/nems.configure.cpld_aero.IN:
```diff
@@ -0,0 +1,128 @@
+OCN_omp_num_threads:            @[ocn_omp_num_threads]
+OCN_attributes::
+  Verbosity = 0
+  DumpFields = @[DumpFields]
```
Same here.

junwang-noaa commented on ush/nems.configure.cpld_aero.IN:
```diff
@@ -0,0 +1,128 @@
+ICE_omp_num_threads:            @[ice_omp_num_threads]
+ICE_attributes::
+  Verbosity = 0
+  DumpFields = @[DumpFields]
```
and here
